operator/autoscaler.go

```diff
@@ -50,86 +50,108 @@ func noopScalingOperation(description string) *ScalingOperation {
 		return NONE
 	}
 
-	status := eds.Status
+	status := as.eds.Status
 
 	// TODO: only consider metric samples that are not too old.
-	sampleSize := len(esMSet.Metrics)
+	sampleSize := len(as.esMSet.Metrics)
 
 	// check for enough data points
-	requiredScaledownSamples := int(math.Ceil(float64(scaling.ScaleDownThresholdDurationSeconds) / metricsInterval.Seconds()))
+	requiredScaledownSamples := int(math.Ceil(float64(scaling.ScaleDownThresholdDurationSeconds) / as.metricsInterval.Seconds()))
 	if sampleSize >= requiredScaledownSamples {
 		// check if CPU is below threshold for the last n samples
 		scaleDownRequired := true
-		for _, currentItem := range esMSet.Metrics[sampleSize-requiredScaledownSamples:] {
+		for _, currentItem := range as.esMSet.Metrics[sampleSize-requiredScaledownSamples:] {
 			if currentItem.Value >= scaling.ScaleDownCPUBoundary {
 				scaleDownRequired = false
 				break
 			}
 		}
 		if scaleDownRequired {
 			if status.LastScaleDownStarted == nil || status.LastScaleDownStarted.Time.Before(time.Now().Add(-time.Duration(scaling.ScaleDownCooldownSeconds)*time.Second)) {
-				log.Infof("EDS %s/%s scaling hint: %s", namespace, name, DOWN)
+				as.logger.Infof("Scaling hint: %s", DOWN)
 				return DOWN
 			}
-			log.Infof("EDS %s/%s not scaling down, currently in cool-down period.", namespace, name)
+			as.logger.Info("Not scaling down, currently in cool-down period.")
 		}
 	}
 
-	requiredScaleupSamples := int(math.Ceil(float64(scaling.ScaleUpThresholdDurationSeconds) / metricsInterval.Seconds()))
+	requiredScaleupSamples := int(math.Ceil(float64(scaling.ScaleUpThresholdDurationSeconds) / as.metricsInterval.Seconds()))
 	if sampleSize >= requiredScaleupSamples {
 		// check if CPU is above threshold for the last n samples
 		scaleUpRequired := true
-		for _, currentItem := range esMSet.Metrics[sampleSize-requiredScaleupSamples:] {
+		for _, currentItem := range as.esMSet.Metrics[sampleSize-requiredScaleupSamples:] {
 			if currentItem.Value <= scaling.ScaleUpCPUBoundary {
 				scaleUpRequired = false
 				break
 			}
 		}
 		if scaleUpRequired {
 			if status.LastScaleUpStarted == nil || status.LastScaleUpStarted.Time.Before(time.Now().Add(-time.Duration(scaling.ScaleUpCooldownSeconds)*time.Second)) {
-				log.Infof("EDS %s/%s scaling hint: %s", namespace, name, UP)
+				as.logger.Infof("Scaling hint: %s", UP)
 				return UP
 			}
-			log.Infof("EDS %s/%s not scaling up, currently in cool-down period.", namespace, name)
+			as.logger.Info("Not scaling up, currently in cool-down period.")
 		}
 	}
 	return NONE
 }
 
 // TODO: check alternative approach by configuring the tags used for `index.routing.allocation`
 // and deriving the indices from there.
-func getScalingOperation(eds *zv1.ElasticsearchDataSet, pods []v1.Pod, direction ScalingDirection, client *ESClient) (*ScalingOperation, error) {
-	esIndices, err := client.GetIndices()
+func (as *AutoScaler) GetScalingOperation() (*ScalingOperation, error) {
```

Comment:
GetScalingOperation is only used in the `package operator`, right? So I wouldn't expose the method.

Reply:
The idea was to indicate that this is actually the entrypoint / public method of the autoscaler. Everything else is private.

Reply:
Ok got it.